addons/point_of_sale/static/src/js/Screens/ProductScreen/ProductScreen.js

```diff
@@ -247,8 +247,8 @@ odoo.define('point_of_sale.ProductScreen', function(require) {
                     let newLine = selectedLine.clone();
                     let decreasedQuantity = currentQuantity - newQuantity
                     newLine.order = order;
-
-                    newLine.set_quantity( - decreasedQuantity);
+                    let keep_price = (newLine.price && decreasedQuantity > 0) ? true : false;
```

Comment:
Do you think the condition still needed? in this case decreasequantity is always higher than 0, and I don't see why there won't be a price. So maybe you can just put "true" as argument in the set_quantity call.

Reply:
thanks for all the suggestions, indeed I double checked and such conditions where not needed anymore

Reply:
thank you for the fix